test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -201,19 +291,29 @@
 (deftest ^:integration legacy-ca-routes-disabled
   (testing (str "(SERVER-759) The legacy CA routes are not forwarded when the "
                 "disabled CA is configured")
-    (logutils/with-test-logging
-      (bootstrap/with-puppetserver-running-with-services-and-mock-jrubies
-       "Mocking JRubies because CA endpoints are pure clojure"
-        app
-       (->> bootstrap/services-from-dev-bootstrap
-            (remove #(= :CaService (tk-services/service-def-id %)))
-            (cons disabled-ca/certificate-authority-disabled-service))
-        {}
-
-        (is (= 404 (:status (http-get "/production/certificate_statuses/all")))
-            (str "A 404 was not returned, indicating that the legacy CA routes "
-                 "are still being forwarded to the core CA functions."))
-
-        (is (not (= 200 (:status (http-get "/production/certificate_statuses/all"))))
-            (str "A 200 was returned from a request made to a legacy CA endpoint "
-                 "indicating the disabled CA service was not detected."))))))
+    ;; Startup server once with real CA service so that SSL certs/keys/etc.
+    ;; are generated properly before we start again with the disabled CA
+    ;; service bootstrapped.  When starting up with disabled CA service,
+    ;; the certs/keys/etc. are assumed to already be in place at startup.  The
+    ;; server would fail to start on being unable to find any of these files.
```

Comment:
Any time you start a webserver with SSL enabled in the configuration, the webserver will need cert files.  When the "real" (non-disabled) CA is configured, it will create its own cert files for the webserver to use.  When the "disabled" CA is configured, it is assumed that you have already dropped the cert files that you want to use in place before starting up the server - so it doesn't need to create its own.

I could have also handled this by having the test create cert files without doing a full service startup or checking some pre-generated certs into the repo for the test to use.  I think we already have too many certs checked into the repo as it is though.  This seemed like a simple enough change and the performance overhead is probably pretty minimal overall.